crates/lox-io/src/ndm/ndm_ci.rs

```diff
@@ -0,0 +1,1326 @@
+                <REF_FRAME>TEME</REF_FRAME>
+                <TIME_SYSTEM>UTC</TIME_SYSTEM>
+                <MEAN_ELEMENT_THEORY>SGP4</MEAN_ELEMENT_THEORY>
+            </metadata>
+            <data>
+                <meanElements>
+                    <EPOCH>2020-12-04T16:27:08.698176</EPOCH>
+                    <MEAN_MOTION>15.31317097</MEAN_MOTION>
+                    <ECCENTRICITY>.0009674</ECCENTRICITY>
+                    <INCLINATION>97.2671</INCLINATION>
+                    <RA_OF_ASC_NODE>51.3486</RA_OF_ASC_NODE>
+                    <ARG_OF_PERICENTER>160.8608</ARG_OF_PERICENTER>
+                    <MEAN_ANOMALY>302.2789</MEAN_ANOMALY>
+                </meanElements>
+                <tleParameters>
+                    <EPHEMERIS_TYPE>0</EPHEMERIS_TYPE>
+                    <CLASSIFICATION_TYPE>U</CLASSIFICATION_TYPE>
+                    <NORAD_CAT_ID>46835</NORAD_CAT_ID>
+                    <ELEMENT_SET_NO>999</ELEMENT_SET_NO>
+                    <REV_AT_EPOCH>436</REV_AT_EPOCH>
+                    <BSTAR>-.13273E-3</BSTAR>
+                    <MEAN_MOTION_DOT>-4.087E-5</MEAN_MOTION_DOT>
+                    <MEAN_MOTION_DDOT>0</MEAN_MOTION_DDOT>
+                </tleParameters>
+            </data>
+        </segment>
+    </body>
+</omm>
+<omm id="CCSDS_OMM_VERS" version="2.0">
+    <header>
+        <CREATION_DATE/>
+        <ORIGINATOR/>
+    </header>
+    <body>
+        <segment>
+            <metadata>
+                <OBJECT_NAME>NUSAT-18 (VERA)</OBJECT_NAME>
+                <OBJECT_ID>2020-079K</OBJECT_ID>
+                <CENTER_NAME>EARTH</CENTER_NAME>
+                <REF_FRAME>TEME</REF_FRAME>
+                <TIME_SYSTEM>UTC</TIME_SYSTEM>
+                <MEAN_ELEMENT_THEORY>SGP4</MEAN_ELEMENT_THEORY>
+            </metadata>
+            <data>
+                <meanElements>
+                    <EPOCH>2020-12-04T13:13:33.140064</EPOCH>
+                    <MEAN_MOTION>15.32037173</MEAN_MOTION>
+                    <ECCENTRICITY>.0009024</ECCENTRICITY>
+                    <INCLINATION>97.2666</INCLINATION>
+                    <RA_OF_ASC_NODE>51.2301</RA_OF_ASC_NODE>
+                    <ARG_OF_PERICENTER>167.2057</ARG_OF_PERICENTER>
+                    <MEAN_ANOMALY>304.5569</MEAN_ANOMALY>
+                </meanElements>
+                <tleParameters>
+                    <EPHEMERIS_TYPE>0</EPHEMERIS_TYPE>
+                    <CLASSIFICATION_TYPE>U</CLASSIFICATION_TYPE>
+                    <NORAD_CAT_ID>46836</NORAD_CAT_ID>
+                    <ELEMENT_SET_NO>999</ELEMENT_SET_NO>
+                    <REV_AT_EPOCH>434</REV_AT_EPOCH>
+                    <BSTAR>.13328E-3</BSTAR>
+                    <MEAN_MOTION_DOT>4.05E-5</MEAN_MOTION_DOT>
+                    <MEAN_MOTION_DDOT>0</MEAN_MOTION_DDOT>
+                </tleParameters>
+            </data>
+        </segment>
+    </body>
+</omm>
+
+</ndm>"#;
+
+        let message: NdmType = from_str(xml).unwrap();
+
+        assert_eq!(
```

Comment:
Well done - these aren't fun to write ><

Reply:
Thanks 😊